pkg/block/index.go

```diff
@@ -347,7 +347,7 @@ func GatherIndexIssueStats(logger log.Logger, fn string, minTime int64, maxTime
 		}
 		l0 := lset[0]
 		for _, l := range lset[1:] {
-			if l.Name <= l0.Name {
+			if l.Name < l0.Name {
```

Comment:
Can we at least warn?

Reply:
You mean warn the user if they are equal? What message do you think would fit here?

Reply:
Yea, for example "Invalid labelset found; multiple labels with the same name for the same series found" 

Wonder how is that even possible? (: I would go even further and put the whole verification step under some flag `accept-malformed-index`, but not sure. (: If any, we can do it in later PRs.

Reply:
Will merge this, let's think about this later on.

Reply:
https://github.com/improbable-eng/thanos/issues/888